internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -185,6 +222,113 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
+// is created for each host (in parallel) using AWS SSM Port Forwarding through
+// a.ssmTarget. Then, each connection is assigned an IP from a.network,
+// starting at the bottom of the IP range. Using iptables, TCP packets destined
+// for the connection's assigned IP are redirected to the connection. Finally,
+// the host's name is mapped to its assigned IP in /etc/hosts.
+func (o *Orchestrator) setupProxyConnections(ctx context.Context, pauseContainer string, a *runTaskAction) error {
+	fmt.Printf("\nSetting up proxy connections...\n")
+
+	ports := make(map[Host]uint16)
+	port := uint16(50000)
+	for i := range a.hosts {
+		ports[a.hosts[i]] = port
+		port++
+	}
+
+	for _, host := range a.hosts {
+		host := host
+		portForHost := ports[host]
+
+		o.wg.Add(1)
+		go func() {
+			defer o.wg.Done()
+
+			err := o.docker.Exec(context.Background(), pauseContainer, io.Discard, "aws", "ssm", "start-session",
+				"--target", a.ssmTarget,
+				"--document-name", "AWS-StartPortForwardingSessionToRemoteHost",
+				"--parameters", fmt.Sprintf(`{"host":["%s"],"portNumber":["%s"],"localPortNumber":["%d"]}`, host.Name, host.Port, portForHost))
+			if err != nil {
+				// report err as a runtime error from the pause container
+				if o.curTaskID.Load() != orchestratorStoppedTaskID {
+					o.runErrs <- fmt.Errorf("proxy to %v:%v: %w", host.Name, host.Port, err)
+				}
+			}
+		}()
+	}
+
+	ip := a.network.IP
+	for host, port := range ports {
+		err := o.docker.Exec(ctx, pauseContainer, io.Discard, "iptables",
+			"--table", "nat",
+			"--append", "OUTPUT",
+			"--destination", ip.String(),
+			"--protocol", "tcp",
+			"--match", "tcp",
+			"--dport", host.Port,
+			"--jump", "REDIRECT",
+			"--to-ports", strconv.Itoa(int(port)))
+		if err != nil {
+			return fmt.Errorf("modify iptables: %w", err)
+		}
+
+		err = o.docker.Exec(ctx, pauseContainer, io.Discard, "iptables-save")
+		if err != nil {
+			return fmt.Errorf("save iptables: %w", err)
+		}
+
+		err = o.docker.Exec(ctx, pauseContainer, io.Discard, "/bin/bash",
+			"-c", fmt.Sprintf(`echo %s %s >> /etc/hosts`, ip.String(), host.Name))
+		if err != nil {
+			return fmt.Errorf("update /etc/hosts: %w", err)
+		}
+
+		ip, err = ipv4Increment(ip, a.network)
+		if err != nil {
+			return fmt.Errorf("increment ip: %w", err)
+		}
+
+		fmt.Printf("Created connection to %v:%v\n", host.Name, host.Port)
+	}
+
+	fmt.Printf("Finished setting up proxy connections\n\n")
+	return nil
+}
+
+// ipv4Increment returns a copy of ip that has been incremented.
+func ipv4Increment(ip net.IP, network *net.IPNet) (net.IP, error) {
+	// make a copy of the previous ip
+	cpy := make(net.IP, len(ip))
+	copy(cpy, ip)
+
+	ipv4 := cpy.To4()
+
+	var inc func(idx int) error
+	inc = func(idx int) error {
+		if idx == -1 {
+			return errors.New("max ipv4 address")
+		}
+
+		ipv4[idx]++
+		if ipv4[idx] == 0 { // overflow occured
+			return inc(idx - 1)
+		}
+		return nil
+	}
+
+	err := inc(3) // 3 since this is an ipv4 address
```

Comment:
nit: instead of adding a comment can we make it a const with example showing why it's 3 there as well?